tests/test_odilo.py

```diff
@@ -0,0 +1,342 @@
+        return data, json.loads(data)
+
+    def error_message(self, error_code, message=None, token=None):
+        """Create a JSON document that simulates the message served by
+        Odilo given a certain error condition.
+        """
+        message = message or self._str
+        token = token or self._str
+        data = dict(errorCode=error_code, message=message, token=token)
+        return json.dumps(data)
+
+
+class TestOdiloCirculationAPI(OdiloAPITest):
+    #################
+    # General tests
+    #################
+
+    # Test 404 Not Found --> patron not found --> 'patronNotFound'
+    def test_01_patron_not_found(self):
+        patron_not_found_data, patron_not_found_json = self.sample_json("error_patron_not_found.json")
+        self.api.queue_response(404, content=patron_not_found_json)
+
+        assert_raises(PatronNotFoundOnRemote, self.api.checkout, '123456789', self.PIN, self.licensepool, 'ACSM')
+        print 'Test patron not found ok!'
+
+    # Test 404 Not Found --> record not found --> 'ERROR_DATA_NOT_FOUND'
+    def test_02_data_not_found(self):
+        data_not_found_data, data_not_found_json = self.sample_json("error_data_not_found.json")
+        self.api.queue_response(404, content=data_not_found_json)
+
+        self.licensepool.identifier.identifier = '12345678'
+        assert_raises(NotFoundOnRemote, self.api.checkout, self.PATRON, self.PIN, self.licensepool, 'ACSM')
+        print 'Test resource not found on remote ok!'
+
+    #################
+    # Checkout tests
+    #################
+
+    # Test 400 Bad Request --> Invalid format for that resource
+    def test_11(self):
+        self.api.queue_response(400, content="")
+        assert_raises(NoAcceptableFormat, self.api.checkout, self.PATRON, self.PIN, self.licensepool, 'FAKE_FORMAT')
+        print 'Test invalid format for resource ok!'
+
+    def test_12_checkout_acsm(self):
+        checkout_data, checkout_json = self.sample_json("checkout_acsm_ok.json")
+        self.api.queue_response(200, content=checkout_json)
+        self.checkout('ACSM')
+
+    def test_13_checkout_ebook_streaming_epub(self):
+        checkout_data, checkout_json = self.sample_json("checkout_ebook_streaming_ok_epub.json")
+        self.api.queue_response(200, content=checkout_json)
+        self.checkout('EBOOK_STREAMING')
+
+    def test_14_checkout_ebook_streaming_pdf(self):
+        self.api.queue_response(400, content="")
+
+        checkout_data, checkout_json = self.sample_json("checkout_ebook_streaming_ok_pdf.json")
+        self.api.queue_response(200, content=checkout_json)
+
+        self.checkout('EBOOK_STREAMING')
+
+    def checkout(self, internal_format):
+        print('Checkout test ' + internal_format)
+
+        loan_info = self.api.checkout(self.PATRON, self.PIN, self.licensepool, internal_format)
+        ok_(loan_info, msg="LoanInfo null --> checkout failed!")
+        print 'Loan ok: %s' % loan_info.identifier
```

Comment:
All the print statements should be removed or converted to log statements, but this one in particular seems misleading.

Reply:
It works because it isn't a "test" method, is an auxiliary method called by the 3 tests above, which queue up the corresponding checkout response each.